src/Chapter2.hs

```diff
@@ -620,7 +650,22 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+
+-- takeEven :: [a] -> [a]
+-- takeEven [] = [] 
+-- takeEven [x] = [x]
+-- takeEven (x:_:xs) = x : takeEven xs
+
+-- takeEven :: [a] -> [a]
+-- takeEven [] = [] 
+-- takeEven (x:xs) = x : takeEven (drop 1 xs)
+
+takeEven :: [a] -> [a]
+takeEven list = go [] list
+    where 
+        go :: [a] -> [a] -> [a]
+        go acc [] = acc
+        go acc (x:xs) = x : (go acc (drop 1 xs))
```

Comment:
You can notice that this implementation doesn't use `acc` at all: you pass an empty list in the beginning, don't change it at all during the recursive calls, and then you return `acc` (which is empty list) in the end.

If you ignore `acc`, you will get exactly the previous implementation but without the recursive call. I think that the pattern with `(x:_:xs)` is the most elegant solution for this problem. When implementing this with the recursive `go`, we had in mind that you will keep a separate `Bool` like `isEven` which would tell you whether this element of a list even or add. So you change the boolean value on each recursive step and decide to prepend element to the result of not depending on this value. Something like that :slightly_smiling_face:

Reply:
Oh! Silly me! But it worked even with an unused accumulator. 
I kept thinking how to use an accumulator but could only find the above solution. 

Thanks for the hint. I have fixed it.